osu.Game.Rulesets.Osu/Skinning/LegacyMainCirclePiece.cs

```diff
@@ -137,19 +135,21 @@ protected override void LoadComplete()
         {
             base.LoadComplete();
 
-            state.BindValueChanged(updateState, true);
             accentColour.BindValueChanged(colour => hitCircleSprite.Colour = LegacyColourCompatibility.DisallowZeroAlpha(colour.NewValue), true);
             if (hasNumber)
                 indexInCurrentCombo.BindValueChanged(index => hitCircleText.Text = (index.NewValue + 1).ToString(), true);
+
+            drawableObject.ApplyCustomUpdateState += updateState;
```

Comment:
Not sure about this direction, feels like we are going to be bitten in the ass by using the bindable in similar cases in the future...

Reply:
Are you able to forego this for the time being since it's a common implementation, and consider an `ArmedState.None` at a later point? There's a lot of things to consider such as how will every ruleset (mods/editor custom states/etc) respond when they receive an `ArmedState.None` event upon `FreeAfterUse()`.

There's a lot of safety checks I want to do when that happens, but I also kinda want to stop working on DHOs and stop breaking changes for a while soon.

Reply:
yeah, but we should probably add a note somewhere about the behaviour (maybe on State's xmldoc? not sure)

Reply:
I've added an xmldoc remark on State for now.